File: skiplist/src/SkipList.java
public interface SkipList<Key extends Comparable<Key>, Value> {
	Value get(Key key);

	void delete(Key key);

	void insert(Key key, Value val);

	void insert(Key[] keys, Value[] vals);

	String toString();

	boolean isEmpty();

	boolean contains(Key key);

	int size();
}

File: vis/src/skiplist/SkipList.ts
import {SkipListNode} from "./SkipListNode";
import {SkipListC} from "./SkipListC";

export interface SkipList{ // interface for skiplist
    get(key: number): GetMethodResult;
    delete(key: number): {}[];
    insert(key: number, val: number): void;
    toString():string;
    isEmpty():boolean;
    size():number;
}

export interface GetMethodResult{ // special type of response for get method
    val: number | null;
    animations:{}[];
}

export interface SearchMethodResult{ // special type of response for search method
    element: SkipListNode | null;
    animations:{}[];
}

export interface animationJson{ // special structure for animation
    c1: SkipListNode | null;
    c2: SkipListNode | null;
    slState: SkipListC;
}